rp2040-hal/src/uart.rs

```diff
@@ -345,40 +396,78 @@ fn configure_baudrate(device: &mut dyn UARTDevice, wanted_baudrate: &Baud, frequ
 
     // PL011 needs a (dummy) line control register write to latch in the
     // divisors. We don't want to actually change LCR contents here.
-    device.uartlcr_h.modify(|_,w| { w });
+    device.uartlcr_h.modify(|_, w| w);
 
-    Ok(Baud((4 * *frequency.integer()) / (64 * baud_ibrd + baud_fbrd) as u32))
+    Ok(Baud(
+        (4 * *frequency.integer()) / (64 * baud_ibrd + baud_fbrd) as u32,
+    ))
 }
 
-
 /// Format configuration. Code loosely inspired from the C SDK.
-fn set_format<'w>(w: &'w mut UART_LCR_H_Writer, data_bits: &DataBits, stop_bits: &StopBits, parity: &Option<Parity>) -> &'w mut UART_LCR_H_Writer {
-
+fn set_format<'w>(
+    w: &'w mut UART_LCR_H_Writer,
+    data_bits: &DataBits,
+    stop_bits: &StopBits,
+    parity: &Option<Parity>,
+) -> &'w mut UART_LCR_H_Writer {
     match parity {
         Some(p) => {
             w.pen().set_bit();
             match p {
                 Parity::Odd => w.eps().bit(false),
-                Parity::Even => w.eps().set_bit()
+                Parity::Even => w.eps().set_bit(),
             };
-        },
-        None => { w.pen().bit(false); }
+        }
+        None => {
+            w.pen().bit(false);
+        }
     };
 
-    unsafe { w.wlen().bits(
-        match data_bits {
-                DataBits::Five => { 0b00 }
-                DataBits::Six => { 0b01 }
-                DataBits::Seven => { 0b10 }
-                DataBits::Eight => { 0b11 }
-            }
-        )
+    unsafe {
+        w.wlen().bits(match data_bits {
+            DataBits::Five => 0b00,
+            DataBits::Six => 0b01,
+            DataBits::Seven => 0b10,
+            DataBits::Eight => 0b11,
+        })
     };
 
     match stop_bits {
         StopBits::One => w.stp2().bit(false),
-        StopBits::Two => w.stp2().set_bit()
+        StopBits::Two => w.stp2().set_bit(),
     };
 
     w
 }
+
+impl<D: UARTDevice> Read<u8> for UARTPeripheral<Enabled, D> {
+    type Error = ReadErrorType;
+
+    fn read(&mut self) -> nb::Result<u8, Self::Error> {
+        let byte: &mut [u8] = &mut [0; 1];
+
+        match self.read_raw(byte) {
+            Ok(_) => Ok(byte[0]),
+            Err(e) => match e {
+                Other(inner) => return Err(Other(inner.err_type)),
```

Comment:
I think the `return`s could be removed since the match would then just resolve to one of the values.

Maybe the two matches could also be merged into one but that is more a matter of taste :smile: